tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/UserEvents/AuthenticationHttpContextExtensionsIntegration.cs

```diff
@@ -0,0 +1,85 @@
+    /// if we make it till here, it means sign in has succeeded
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = AssemblyName,
+        TypeName = HttpContextExtensionsTypeName,
+        ParameterTypeNames = new[] { "Microsoft.AspNetCore.Http.HttpContext", ClrNames.String, "System.Security.Claims.ClaimsPrincipal", "Microsoft.AspNetCore.Authentication.AuthenticationProperties" },
+        MethodName = "SignInAsync",
+        ReturnTypeName = "System.Task",
+        MinimumVersion = Major2,
+        MaximumVersion = Major7,
+        IntegrationName = nameof(IntegrationId.AspNetCore))]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static class AuthenticationHttpContextExtensionsIntegration
+    {
+        private const string Major2 = "2";
+        private const string Major7 = "7";
+        private const string AssemblyName = "Microsoft.AspNetCore.Authentication.Abstractions";
+
+        private const string HttpContextExtensionsTypeName = "Microsoft.AspNetCore.Authentication.AuthenticationHttpContextExtensions";
+
+        internal static CallTargetState OnMethodBegin<TTarget>(object httpContext, string scheme, ClaimsPrincipal claimPrincipal, object authProperties)
+        {
+            var security = Security.Instance;
+            if (security.TrackUserEvents)
+            {
+                var tracer = Tracer.Instance;
+                var scope = tracer.InternalActiveScope;
+                return new CallTargetState(scope, claimPrincipal);
+            }
+
+            return CallTargetState.GetDefault();
+        }
+
+        internal static object OnAsyncMethodEnd<TTarget>(object returnValue, Exception exception, in CallTargetState state)
+        {
+            var claimsPrincipal = state.State as ClaimsPrincipal;
+            if (claimsPrincipal?.Claims != null && Security.Instance is { TrackUserEvents: true } security)
+            {
+                var span = state.Scope.Span;
+                var setTag = TaggingUtils.GetSpanSetter(span, out _);
+                var tryAddTag = TaggingUtils.GetSpanSetter(span, out _, replaceIfExists: false);
+                setTag(Tags.AppSec.EventsUsers.LoginEvent.SuccessTrack, "true");
+                setTag(Tags.AppSec.EventsUsers.LoginEvent.SuccessAutoMode, Security.Instance.Settings.UserEventsAutomatedTracking);
+                foreach (var claim in claimsPrincipal.Claims)
+                {
+                    if (claim.Type == ClaimTypes.NameIdentifier)
+                    {
+                        if (Guid.TryParse(claim.Value, out _))
+                        {
+                            tryAddTag(Tags.User.Id, claim.Value);
+                            break;
+                        }
+                    }
+
+                    // todo in extended mode, email, username, login (if not available, see signin integrations
+                }
```

Comment:
Unfortunately, the name of the user won't _necessarily_ be set to `ClaimTypes.NameIdentifier`. In fact, for APIs, it'll rarely be this. `ClaimTypes.NameIdentifier` is one of MS's horrible SOAPy constants: `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier`

I'd suggest taking a slightly different approach and using `ClaimsPrincipal.Identity?.Name` instead. 

Then this becomes something like:

```suggestion
                if (claimsPrincipal.Identity?.Name { } name && Guid.TryParse(name, out _))
                {
                    tryAddTag(Tags.User.Id, claim.Value);
                }

                // todo in extended mode, email, username, login (if not available, see signin integrations
```

Alternatively, you can try also looking for other common field claims used by like `"sub"`  (which should probably take precedence to be honest, as this by definition has to be a unique ID so is often an ID, but sometimes is an email). And there's `name`. See this list basically: https://learn.microsoft.com/en-us/dotnet/api/system.identitymodel.tokens.jwt.jwtregisteredclaimnames?view=msal-web-dotnet-latest

🙂